src/test/java/ru/ayaz/UserMessageHandlerTest.java

```diff
@@ -0,0 +1,123 @@
+package ru.ayaz;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import static junit.framework.TestCase.assertEquals;
+import static junit.framework.TestCase.assertTrue;
+import static org.mockito.Mockito.*;
+
+public class UserMessageHandlerTest {
+
+    private UserMessageHandler messageHandler;
+
+    @Before
+    public final void before() {
+        messageHandler = new UserMessageHandler(500);
+    }
+
+    @Test
+    public void shouldBeAbleToEnqueueMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+
+        messageHandler.enqueueMessage(someMessage);
+
+        assertTrue(messageHandler.queueContainsMessage(someMessage));
+    }
+
+    @Test
+    public void shouldBeAbleToGetEnqueuedMessage() throws InterruptedException {
+        UserMessage someMessage = new UserMessage("Ayaz", "Hello");
+        messageHandler.enqueueMessage(someMessage);
+
+        UserMessage gottenMessage = messageHandler.takeMessage();
+
+        assertEquals(someMessage, gottenMessage);
+    }
+
+
+    @Test
+    public void shouldInvokePrintUnknownCommandWhenMalformedCommandTaken() throws InvalidNicknameException, InterruptedException {
```

Comment:
&#x1F34E; **Issue**
You probably want to just throw "Exception" in each your @Test method, so that you wouldn't have to write these ugly long lists of possible exceptions your method could throw.